File: content-solutions/stack/snowball-merge.js
function solution(sizes) {
  const stack = [];
  for (let ball of sizes) {
    while (stack.length > 0 && stack[stack.length - 1] === ball) {
      stack.pop();
      ball *= 2;
    }
    stack.push(ball);
  }
  return stack;
}

File: content-solutions/stack/snowball-merge.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] sizes) {
        Deque<Integer> stack = new ArrayDeque<>();
        for (int size : sizes) {
            int ball = size;
            while (!stack.isEmpty() && stack.peek() == ball) {
                stack.pop();
                ball *= 2;
            }
            stack.push(ball);
        }
        List<Integer> answer = new ArrayList<>(stack);
        Collections.reverse(answer);
        return answer;
    }
}
